src/Surfnet/StepupGateway/GatewayBundle/Sso2fa/CookieService.php

```diff
@@ -121,17 +119,13 @@ public function handleSsoOn2faCookieStorage(
                     $secondFactor->institution
                 )
             );
+
             if ($isEnabled) {
-                // The cookie reader can return a NullCookie if the cookie is not present, was expired or was otherwise
-                // deemed invalid. See the CookieHelper::read implementation for details.
-                $ssoCookie = $this->read($request);
                 $identityId = $responseContext->getIdentityNameId();
                 $loa = $secondFactor->getLoaLevel($this->secondFactorTypeService);
-                $isValid = $this->isCookieValid($ssoCookie, $loa, $identityId);
                 $isVerifiedBySsoOn2faCookie = $responseContext->isVerifiedBySsoOn2faCookie();
-                // Did the LoA requirement change? If a higher LoA was requested, or did a new token authentication
-                // take place? In that case create a new SSO on 2FA cookie
-                if ($this->shouldAddCookie($ssoCookie, $isValid, $loa, $isVerifiedBySsoOn2faCookie)) {
+                // Did the user perform a new second factor authentication?
+                if (!$isVerifiedBySsoOn2faCookie) {
```

Comment:
For now, The only two options are token or cookie, but this might change in the future. 
Can we create a ```$isVerifiedByTokenAuthentication``` to prevent mistakes in the future?

Reply:
That crossed my mind. But left it like this for now. As we explicitly alter the state when the sso cookie auth took place. And not when the regular 2FA authentication takes place. I know this is a matter of perspective. But I dont think adding an inverse will create a less confusing situation later down the road. Or am I overlooking something?